File: 백준/Silver/11399. ATM/ATM.py
import sys

N = int(sys.stdin.readline())
time = list(map(int, sys.stdin.readline().split(" ")))
time.sort()

answer = [0 for _ in range(N)]
answer[0] = time[0]
sum = answer[0]

for i in range(1, N):
    answer[i] = answer[i-1]+time[i]
    sum += answer[i]

print(sum)

File: 백준/Silver/11399. ATM/ATM.go
package main

import (
	"os"
	"bufio"
	"fmt"
	"sort"
)

func main(){
	reader :=bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	var size int

	fmt.Fscanln(reader, &size)
	arr := make([]int, size)

	for i := 0; i < size; i++ {
		fmt.Fscan(reader, &arr[i])
	}

	sort.Ints(arr)
	time:=0
	sum := 0

	for i := 0; i < size; i++ {
		time = time + arr[i]
		sum += time
	}

	defer writer.Flush()
	fmt.Fprintln(writer, sum)
}